jpa/hibernate-transformer/src/main/java/org/jboss/as/hibernate/Hibernate51CompatibilityTransformer.java

```diff
@@ -360,12 +192,28 @@ private static String replaceSessionImplementor(String desc) {
                 "Lorg/hibernate/engine/spi/SharedSessionContractImplementor;");
     }
 
+    private boolean parentExpectsSharedSessionContractImplementor(Set parentClassesAndInterfaces) {
+        return (parentClassesAndInterfaces.contains("org/hibernate/usertype/UserType")
+                || parentClassesAndInterfaces.contains("org/hibernate/usertype/CompositeUserType")
+                || parentClassesAndInterfaces.contains("org/hibernate/usertype/UserCollectionType")
+                || parentClassesAndInterfaces.contains("org/hibernate/usertype/UserVersionType")
+                || parentClassesAndInterfaces.contains("org/hibernate/type/Type")
+                || parentClassesAndInterfaces.contains("org/hibernate/type/SingleColumnType")
+                || parentClassesAndInterfaces.contains("org/hibernate/type/AbstractStandardBasicType")
+                || parentClassesAndInterfaces.contains("org/hibernate/type/ProcedureParameterExtractionAware")
+                || parentClassesAndInterfaces.contains("org/hibernate/type/ProcedureParameterNamedBinder")
+                || parentClassesAndInterfaces.contains("org/hibernate/type/VersionType")
+                || parentClassesAndInterfaces.contains("org/hibernate/collection/spi/PersistentCollection"));
+    }
+
     private void collectClassesAndInterfaces(Set<String> classesAndInterfaces, ClassLoader classLoader, String className) {
         if (className == null || "java/lang/Object".equals(className)) {
             return;
         }
 
-        try (InputStream is = classLoader.getResourceAsStream(className.replace('.', '/') + ".class")) {
+        InputStream is = null;
+        try {
+            is = classLoader.getResourceAsStream(className.replace('.', '/') + ".class");
```

Comment:
Is there a reason the try-with-resources was replaced a finally? Not a big deal, just seems unnecessary.

Reply:
I thought it looked leaky, I agree, try-with-resources is better.

Reply:
reverted back to the try-with-resources